From fd2ec0b1eb5713124161da2934964769290eb2e4 Mon Sep 17 00:00:00 2001
From: Ludovic LANGE <github@lange.nom.fr>
Date: Sat, 16 Aug 2014 15:10:56 +0200
Subject: [PATCH 04/10] Fix for #647 - GUI Tabs do not switch on WX GUI Scope
 sink channels on OSX

With this patch, the bug is not showing anymore. It was only tested by
me on Mac OS X - not on Linux nor other platforms.
The possible regressions introduced by fix should be limited to the
notebook chooser forms only.
---
 gr-wxgui/python/wxgui/forms/forms.py |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/gr-wxgui/python/wxgui/forms/forms.py b/gr-wxgui/python/wxgui/forms/forms.py
index db55472..f819f13 100644
--- a/gr-wxgui/python/wxgui/forms/forms.py
+++ b/gr-wxgui/python/wxgui/forms/forms.py
@@ -515,13 +515,13 @@ class notebook(_chooser_base):
 		_chooser_base.__init__(self, **kwargs)
 		assert len(pages) == len(self._choices)
 		self._notebook = notebook
-		self._notebook.Bind(wx.EVT_NOTEBOOK_PAGE_CHANGED, self._handle)
+		self._notebook.Bind(wx.EVT_NOTEBOOK_PAGE_CHANGING, self._handle)
 		#add pages, setting the label on each tab
 		for i, page in enumerate(pages):
 			self._notebook.AddPage(page, self._labels[i])
 		self._add_widget(self._notebook)
 
-	def _handle(self, event): self[INT_KEY] = self._notebook.GetSelection()
+	def _handle(self, event): self[INT_KEY] = event.GetSelection()
 	# SetSelection triggers a page change event (deprecated, breaks on Windows) and ChangeSelection does not
 	def _update(self, i): self._notebook.ChangeSelection(i)
 
-- 
1.7.10.4

